client/my-sites/customer-home/main.jsx

```diff
@@ -49,6 +49,18 @@ const Home = ( {
 	const translate = useTranslate();
 	const reduxDispatch = useDispatch();
 
+	useEffect( () => {
+		if ( ! canUserUseCustomerHome || ! layout || noticeType !== 'purchase-success' ) {
+			return;
+		}
+		const successMessage = translate( 'Your purchase has been completed!' );
+		reduxDispatch(
+			successNotice( successMessage, {
+				isPersistent: true,
+			} )
+		);
+	}, [ noticeType, layout, reduxDispatch, translate ] );
```

Comment:
Minor: I think `canUserUseCustomerHome` needs to be in the dependencies array for this (possibly helpful: if you set up your editor to show eslint warnings, this should show up) since that's used in the block. It _probably_ won't matter (I imagine that variable won't change without other variables changing too) but it's nice not to have extra warnings in the code.

Reply:
So another way of approaching this is that we can dispatch a global notice as a side-effect. This can be done from this specific checkout data handling, or as a side effect of the [notice](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/notices/middleware.js) or [navigation](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/navigation/middleware.js) middleware. (Generally. I'd try the former first, especially if the action isn't general enough to warrant middleware handling.)

If you're interested, let me know if you need a working example. @dmsnell is also usually available for advice on data-layer handling.

Reply:
FWIW we're currently in the process of removing the notice middleware completely. I'd advise not introducing new notices that are implemented with it. Let me know if you need additional details or rationale.

Reply:
Thanks @tyxla Are we still supporting more complex side effects in the data-layer folder https://github.com/Automattic/wp-calypso/tree/trunk/client/state/data-layer/wpcom

That's what I'd try first for showing a success notice after a purchase.

Reply:
@gwwar Sure, dispatching a notice that corresponds to an action type in the data layer is just fine.  👍 

The primary reason for us to want to remove the notices middleware is that it prevents state from being fully modularized. Data layer, on the other hand, is modular already, so as long as we don't rely on that functionality at boot time, we're good.

Reply:
Thanks @tyxla, and 🚀 for the data modularization efforts!